client/app/services/group/currentGroup.service.js

```diff
@@ -14,7 +14,18 @@ export default class CurrentGroup {
     Object.assign(this, {
       value: {},
       User,
-      SessionUser
+      SessionUser,
+      map: {
```

Comment:
Yeah, I think I get what you are saying with "single observer". And having data (vs. methods) also makes it easier if somehow storeDetail loads faster than groupMap. So I just changed it to use only data in the CurrentGroup now, and also removed the timeouts (not quite sure why it didn't even display the markers before, should see if we report the problem again sometime...)